### PR TITLE
[V1.0.0] Roles and RoleGuard

### PR DESCRIPTION
**Pull request by Milutin-Stan**
![][trello.png] [**[v1.0.0] Mettre en place les rôles (Admin Extra Employeur)**](https://trello.com/c/84bNoN4O/13-v100-mettre-en-place-les-r%C3%B4les-admin-extra-employeur)

# Description

Added roles (Admin, Extra and Employeur) with updated services and tests.
A new RoleGuard was created to protect routes for the corresponding roles (The RoleGuard also uses AuthGuard within it so that you dont need to use both).
The coverage config was also changed.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

[trello.png]: https://github.trello.services/images/mini-trello-icon.png

### CODE REVIEW

File: src/domain/utils/enums/role.enum.ts
Comment (DaFonsecaDamien) [inline src/domain/utils/enums/role.enum.ts:1]: Mettre les champs en Anglais

File: src/domain/services/role/role.service.ts
Comment (DaFonsecaDamien) [inline src/domain/services/role/role.service.ts:1]: Ajouter une méthode pour permettre la suppression d'un role en base

File: src/application/user/user.controller.ts
Comment (DaFonsecaDamien) [inline src/application/user/user.controller.ts:1]: Séparer la logique de vérification de l'authentification et du rôle.
Entre autre mettre l'annotation (@JWTGuard  ou autre nom) puis celle du rôle, afin que le rôle Guard fasse une et unique chose.

File: src/domain/guards/role.guard.ts
Comment (DaFonsecaDamien) [inline src/domain/guards/role.guard.ts:1]: Laisser le RolesGuard tester uniquement les rôles et non pas l'authentification.

**Review by DaFonsecaDamien — CHANGES_REQUESTED** [review]
Réaliser les changements demandés dans les commentaires avant le merge.

**Review by DaFonsecaDamien — APPROVED** [review]
Nice work 👍🏼

File: src/domain/utils/enums/role.enum.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum RoleEnum {
+  ADMIN = "ADMIN",
+  EXTRA = "EXTRA",
+  EMPLOYEUR = "EMPLOYEUR",
```
Comment (AdrienRoux) [inline src/domain/utils/enums/role.enum.ts]:
```suggestion
  EMPLOYEUR = "EMPLOYER",
```

**Review by AdrienRoux — CHANGES_REQUESTED** [review]
Minor change

**Review by AdrienRoux — APPROVED** [review]
Fine for me ! :)

**Pull request merged**